### PR TITLE
PLA-3742: responsive widget

### PR DESCRIPTION
**Pull request by LoMatzke**
TICKET:
https://heycater.atlassian.net/jira/software/c/projects/PLA/boards/2?modal=detail&selectedIssue=PLA-3742

Screenshot 2022-11-25 at 14 39 10

### CODE REVIEW

**Review by wivern-co-uk — DISMISSED** [review]
Not all ACs implemented. @LoMatzke please self-check on every ticket, this should be a habbit

File: src/App.js [hunk 1]
```diff
@@ -16,23 +16,23 @@ function App({ domElement }) {
...
 
   return (
-    <div
-      onClick={() =>
-        (window.location.href = `https://www.heycater.com/de/catering/${city}/${catererSlug}?utm_source=online-shop-widget`)
-      }
-      className={`shop__button__heycater_app ${position}`}
+    <a
+      href={`https://www.heycater.com/de/catering/${city}/${catererSlug}?utm_source=online-shop-widget`}
+      target="top"
```
Comment (wivern-co-uk) [inline src/App.js]: @LoMatzke please check `target="top"` is a correct attr to jump out of iframes

File: src/App.js [hunk 2]
```diff
@@ -16,23 +16,23 @@ function App({ domElement }) {
...
-      <div className="heycater__widget_container">
-        <button className="heycater__widget_cta">
-          <div className="heycater__widget_cart">
-            <Cart />
-          </div>
-          <div className="heycater__widget_button-text">
-            <Logo /> Online buchen
-          </div>
-        </button>
+      <div className={`shop__button__heycater_app ${position}`}>
```
Comment (wivern-co-uk) [inline src/App.js]: Looks like we can turn this `div` into `a` to reduce amount of elements

File: src/App.js [hunk 3]
```diff
@@ -16,23 +16,23 @@ function App({ domElement }) {
...
+        <div className="heycater__widget_container">
+          <button className="heycater__widget_cta">
```
Comment (wivern-co-uk) [inline src/App.js]: `button` couldn't be inside `a`, do we really need it here?

File: dist/index.css [hunk 4]
```diff
@@ -1 +1 @@
...
+.shop__button__heycater_app{font-family:Arial,sans-serif;position:fixed;cursor:pointer}.heycater__widget_placement__bottom{bottom:10px;right:10px}.heycater__widget_placement__middle{bottom:50%;right:10px}.heycater__widget_placement__top{top:10px;right:10px}.heycater__widget_container,.heycater__widget_cta{display:flex;align-items:center;justify-content:center;flex-direction:row;cursor:pointer}.heycater__widget_cta{width:80px;height:80px;border-radius:100px;border:2px solid #e86202;background-color:#e86202;color:#fff;transition:width .7s;box-shadow:2px 2px rgba(0,0,0,.1)}@media (max-width:600px){.heycater__widget_cta{width:40px;height:40px}}.heycater__widget_cta .heycater__widget_button-text{opacity:1;transition:display .5s;width:100%;left:0;display:none;white-space:nowrap;font-size:16px;cursor:pointer}.heycater__widget_cta:hover{width:180px;transition:width .7s;cursor:pointer}.heycater__widget_cart{margin-top:12px;margin-left:3px;cursor:pointer}.heycater__widget_cta:hover .heycater__widget_cart{display:none;cursor:pointer}.heycater__widget_cta:hover .heycater__widget_button-text{opacity:1;display:flex;align-items:center;transition:opacity .5s;cursor:pointer}
```
Comment (wivern-co-uk) [inline dist/index.css]: @javiersuazo any chance we can move css into js file, so widget could load faster in non-render-blocking way?

File: src/App.js (hunk 1 above)
Comment (LoMatzke) [reply src/App.js]: @wivern-co-uk the target="_top" seems correct, just my syntax was not

_The [target](https://www.dofactory.com/html/target) attribute specifies where the linked page or form response will be opened.The target="_top" value opens the page in the top frame which always is the full browser tab/window._

Comment (LoMatzke) [reply src/App.js]: @wivern-co-uk question to me is if we want to open the MP in the same window or open new window?

File: dist/index.css (hunk 4 above)
Comment (javiersuazo) [reply dist/index.css]: We can, I would address it on a different ticket tho. We either upgrade to parcel 2 or we use a library like style components. Just to not increase complexity on this ticket.

**Review by javiersuazo — APPROVED** [review]

**Review by thezainsaleem — APPROVED** [review]

File: src/App.js (hunk 3 above)
Comment (wivern-co-uk) [reply src/App.js]: Upvoted, marked as resolved, but in fact - unresolved.

Comment (LoMatzke) [reply src/App.js]: @wivern-co-uk all fixed now, I had unpushed changes.

**Review by javiersuazo — APPROVED** [review]

**Review by wivern-co-uk — APPROVED** [review]

**Pull request merged**